### ISSUE TITLE
Add response trailer to simulation schema

### ISSUE THREAD

**Issue by tommysitu** [SpectoLabs/hoverfly#828]
**Is your feature request related to a problem? Please describe.**
Some streaming API contains Trailer after the chunked response body. Those Trailers are not captured, hence cannot be simulated by Hoverfly.

More info about Trailers:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers
http://engineering.pivotal.io/post/http-trailers/

**Describe the solution you'd like**
Add a `trailers` field to the response object in the simulation schema.
```
 "response" : {
        "status" : 200,
        "body" : "AAAAAAA=",
        "encodedBody" : true,
        "templated" : false,
        "headers" : {
          "Date" : [ "Fri, 22 Mar 2019 16:44:17 GMT" ],
          "Hoverfly" : [ "Was-Here" ]
        },
        "trailers" : {
          "X-Streaming-Error" : [ "Connection Closed" ]
        }
      }
```

**Describe alternatives you've considered**
Storing the trailers in the headers field? But trailers must be written out at the end of the chunked body. We need a way to distinguish the trailers if they are stored in the header fields, like adding a unique prefix to the name. 

**Additional context**
N/A

**Comment by tommysitu** [comment]
Another workaround is to record the trailers in the `headers` array: 

```
"response" : {
    "headers" : {
           "Trailer": ["X-Streaming-Error"],
           "X-Streaming-Error" : [ "Connection Closed" ]
     }
}
```

`Trailer` declares the trailer keys. The headers with those keys will be sent in the response trailer instead. It's similar to how `ResponseWriter` set trailers in Golang.

**Comment by tommysitu** [comment]
Fixed in v1.0.0
